appservice/src/createAppService/IAppServiceWizardContext.ts

```diff
@@ -58,4 +58,11 @@ export interface IAppServiceWizardContext extends IResourceGroupWizardContext, I
      * By specifying this in the context, we can ensure that Azure is only queried once for the entire wizard
      */
     plansTask?: Promise<AppServicePlan[]>;
+
+    /**
+     * The runtime to put to the top of the QuickPick list to recommend to the user.
+     * This should be set in `setWizardContextDefaults`
+     */
+
+     detectedSiteRuntime?: string;
```

Comment:
Maybe `recommendedSiteRuntime`? I feel like that more closely matches your comment above. "detected" is more about "how you picked this value" and I think the name should be more about "how you're going to use this value"